routing/routing_session.hpp

```diff
@@ -35,7 +38,19 @@ struct SpeedCameraRestriction
   SpeedCameraRestriction() : m_index(0), m_maxSpeedKmH(numeric_limits<uint8_t>::max()) {}
 };
 
-class RoutingSession
+class TrafficCache
```

Comment:
I moved TrafficInforGetter interface and TrafficCache implementation to a separate hpp/cpp unit in traffic subsystem.

To get rid of TrafficInfoGetterNoJam/TrafficInfoGetterForTesting I need to inherit RoutingSession from TrafficInfoGetter (you declared above) which is not an interface. Besides that I need to pass TrafficInfoGetter(you declared above) to EdgeEstimator::CreateForCar() while EdgeEstimator needs only Get() method.

Don't you think it's a to big price for getting rid of TrafficInfoGetterNoJam/TrafficInfoGetterForTesting?